### PR TITLE
Register login

### PR DESCRIPTION
**Pull request by tomaszleszczynski1980**
Added register form, added new table to models, added route to app.py

### CODE REVIEW

File: app.py [hunk 1]
```diff
@@ -40,16 +41,14 @@ def add_event():
     """
     form = AddEventForm(request.form)
     available_categories = Category.query.order_by("title")
-    form.event_categories.choices = [
-        (c.id, c.title) for c in available_categories
-    ]
+    form.event_categories.choices = [(c.id, c.title) for c in available_categories]
```
Comment (kMatejak) [inline app.py]: now is longer than 80

File: app.py [hunk 2]
```diff
@@ -77,14 +76,29 @@ def edit_event(int: id):
...
 
-@app.route('/register', methods = ['GET', 'POST'])
+# @app.route('/register', methods = ['GET', 'POST'])
+# def register():
+#     if request.method == 'POST':
+#         request.form['']
+#         return redirect('index')
+#     else:
+#         return render_template('users/register_old.html',
+#                                 title='KtoNadaje',)
```
Comment (kMatejak) [inline app.py]: it's better to delete it than to shade it

File: app.py [hunk 3]
```diff
@@ -77,14 +76,29 @@ def edit_event(int: id):
...
+    if request.method == 'POST' and form.validate():
+        user = User(username=form.username.data,
+                    email=form.email.data,
+                    password=dp.password_hash(form.password.data)
+                    )
+        db.session.add(user)
+        db.session.commit()
+        # flash('Registration succesfull')
+        return redirect(url_for('index'))
+    return render_template('users/register.html', form=form)
```
Comment (kMatejak) [inline app.py:106]: and is that works on your dev environment? I don't have time to check what's wrong with migrations (I can't upgrade my database (alembic issue))

File: decrypt_password.py [hunk 4]
```diff
@@ -0,0 +1,10 @@
+import bcrypt as bc
+
+def password_hash(plain_password: str):
+
+    hashed_password = bc.hashpw(plain_password.encode('utf-8'), bc.gensalt())
+    return hashed_password.decode('utf-8')
+
+
+def verify_password(plain_password: str, hashed_password: str):
+    return bc.checkpw(plain_password.encode('utf-8'), hashed_password.encode('utf-8'))
```
Comment (kMatejak) [inline decrypt_password.py]: add one empty line at the bottom of the file

**Comment by kMatejak** [comment]
I have good sources to think about the topic of keeping passwords:
-(shorter one) https://itnext.io/how-not-to-store-passwords-4955569e6e84 
- (that longest one) https://nakedsecurity.sophos.com/2013/11/20/serious-security-how-to-store-your-users-passwords-safely/

**Comment by kMatejak** [comment]
I have good sources to think about the topic of keeping passwords:
[shorter one](https://itnext.io/how-not-to-store-passwords-4955569e6e84)
[that longer one](https://nakedsecurity.sophos.com/2013/11/20/serious-security-how-to-store-your-users-passwords-safely/)

**Comment by kMatejak** [comment]
I was thinking about not keeping passwords in the database at all, but Wiktor told me he didn't want to login via Facebook / Twitter / Gmail.

**Comment by kMatejak** [comment]
[Here is a good (and short) reference for implementing password hashing with bcrypt](https://uniwebsidad.com/libros/explore-flask/chapter-12/storing-passwords)

**Comment by kMatejak** [comment]
I would think about using [Flask Security](https://pythonhosted.org/Flask-Security/), but I don't think we have time for this.

File: app.py (hunk 1 above)
Comment (tomaszleszczynski1980) [reply app.py]: I changed it just to understand. We can change it back.

File: app.py (hunk 2 above)
Comment (tomaszleszczynski1980) [reply app.py]: I do not delete anything yet.

File: app.py (hunk 3 above)
Comment (tomaszleszczynski1980) [reply app.py:106]: Works.

File: decrypt_password.py (hunk 4 above)
Comment (tomaszleszczynski1980) [reply decrypt_password.py]: ok

File: app.py (hunk 1 above)
Comment (kMatejak) [reply app.py]: ok. change it back, please

File: app.py (hunk 2 above)
Comment (kMatejak) [reply app.py]: delete it :)

File: app.py (hunk 3 above)
Comment (kMatejak) [reply app.py:106]: ok

**Review by kMatejak — APPROVED** [review]

**Pull request merged**